dhall-docs/src/Dhall/Docs/Markdown.hs

```diff
@@ -0,0 +1,29 @@
+{-| Contains all utilities related to markdown processing
+-}
+module Dhall.Docs.Markdown (
+    markdownToHtml
+) where
+
+import Data.Text       (Text)
+import Lucid           (Html)
+import Path            (Abs, File, Path)
+import Text.Megaparsec (ParseErrorBundle (..))
+import Text.MMark      (MMarkErr)
+
+import qualified Lucid
+import qualified Path
+import qualified Text.MMark as MMark
+
+type ParseError = ParseErrorBundle Text MMarkErr
+
+{-| Takes a text that could contain markdown and returns the generated HTML.
+    If an error occurs while parsing, it also returns the error information.
+-}
+markdownToHtml
+    :: Path Abs File -- ^ Used by `Mmark.parse` for error messages
+    -> Text          -- ^ Text to parse
+    -> (Maybe ParseError, Html ())
```

Comment:
I think it might be better to return `Either ParseError (Html ()))` here.

The current type is a bit confusing IMHO. When we get a parse error, the returned HTML is _not_ generated from the parsed markdown, contrary to what the function name seems to say!